helm-chart/binderhub/templates/_helpers.tpl

```diff
@@ -36,11 +36,11 @@ Render docker config.json for the registry-publishing secret.
 {{- end }}
 {{- $username := .Values.registry.username -}}
 
-{
-  "auths": {
-    "{{ $url }}": {
-      "auth": "{{ printf "%s:%s" $username .Values.registry.password | b64enc }}"
-    }
-  }
-}
+{{- /* prepare a dict containing the registry credentials */ -}}
+{{- $dockerConfig := dict "auths" (dict $url (dict "auth" (printf "%s:%s" $username .Values.registry.password | b64enc))) }}
+{{- if .Values.config.BinderHub.buildDockerConfig }}
+{{- $dockerConfig := merge $dockerConfig $dockerConfig .Values.config.BinderHub.buildDockerConfig }}
+{{- end }}
+
+{{ $dockerConfig | toPrettyJson }}
```

Comment:
I wanted to verify that this template renders what we already did, and it does except for some blank lines. This suggestion makes everything render like before.

```suggestion
{{- /* initialize a dict to represent a docker config with registry credentials */}}
{{- $dockerConfig := dict "auths" (dict $url (dict "auth" (printf "%s:%s" $username .Values.registry.password | b64enc))) }}

{{- if .Values.config.BinderHub.buildDockerConfig }}
{{- $dockerConfig := merge $dockerConfig $dockerConfig .Values.config.BinderHub.buildDockerConfig }}
{{- end }}

{{- $dockerConfig | toPrettyJson }}
```